crates/orderbook/src/arguments.rs

```diff
@@ -137,6 +137,10 @@ pub struct Arguments {
     /// Max number of limit orders per user.
     #[clap(long, env, default_value = "10")]
     pub max_limit_orders_per_user: u64,
+
+    /// Enable buy ETH orders paying to smart contract wallets.
+    #[clap(long, env, default_value = "false")]
+    pub enable_smart_contract_payments: bool,
```

Comment:
tiny naming nit:
If only orders having `ETH` as the `buy_token` are a problem then `enable_smart_contract_payments` seems too general.
Maybe `enable_eth_payments_to_smart_contracts`?